test/spec/modules/conversantAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,733 @@
+        )
+      ).to.be.true;
+      conversantAnalytics.disableAnalytics();
+    });
+
+    it('should sample when sampling set to 1', function() {
+      sandbox.stub(utils, 'logError');
+
+      VALID_CONFIGURATION.options.sampleRate = 100;
+      conversantAnalytics.enableAnalytics(VALID_CONFIGURATION);
+      expect(utils.logError.called).to.equal(false);
+      expect(caObj.DO_SAMPLE).to.equal(true);
+      conversantAnalytics.disableAnalytics();
+      delete VALID_CONFIGURATION.options.sampleRate;
+    });
+
+    it('should NOT sample when sampling set to 0', function() {
+      sandbox.stub(utils, 'logError');
+
+      VALID_CONFIGURATION.options.sampleRate = 0;
+      conversantAnalytics.enableAnalytics(VALID_CONFIGURATION);
+      expect(utils.logError.called).to.equal(false);
+      expect(caObj.DO_SAMPLE).to.equal(false);
+      conversantAnalytics.disableAnalytics();
+      delete VALID_CONFIGURATION.options.sampleRate;
+    });
+  });
+
+  describe('Helper Function Tests', function() {
+    it('should cleanup up cache objects', function() {
+      conversantAnalytics.enableAnalytics(VALID_CONFIGURATION);
+
+      caObj.adIdLookup['keep'] = {timeReceived: DATESTAMP + 1};
+      caObj.adIdLookup['delete'] = {timeReceived: DATESTAMP - caObj.MAX_MILLISECONDS_IN_CACHE};
+
+      caObj.timeoutCache['keep'] = {timeReceived: DATESTAMP + 1};
+      caObj.timeoutCache['delete'] = {timeReceived: DATESTAMP - caObj.MAX_MILLISECONDS_IN_CACHE};
+
+      expect(Object.keys(caObj.adIdLookup).length).to.equal(2);
+      expect(Object.keys(caObj.timeoutCache).length).to.equal(2);
+
+      clock.tick(caObj.CACHE_CLEANUP_TIME_IN_MILLIS);
+      expect(Object.keys(caObj.adIdLookup).length).to.equal(1);
+      expect(Object.keys(caObj.timeoutCache).length).to.equal(1);
+
+      conversantAnalytics.disableAnalytics();
+
+      // After disable we should cleanup the cache
+      expect(Object.keys(caObj.adIdLookup).length).to.equal(0);
+      expect(Object.keys(caObj.timeoutCache).length).to.equal(0);
+    });
+
+    it('isInt() should return true', function() {
+      expect(caObj.isInt(1)).to.equal(true);
+      expect(caObj.isInt(0)).to.equal(true);
+      expect(caObj.isInt(-1)).to.equal(true);
+      expect(caObj.isInt(111)).to.equal(true);
+      expect(caObj.isInt(1.0)).to.equal(true);
+    });
+
+    it('isInt() should return false', function() {
+      expect(caObj.isInt(0.1)).to.equal(false);
+      expect(caObj.isInt(1.1)).to.equal(false);
+      expect(caObj.isInt('foo')).to.equal(false);
+      expect(caObj.isInt(undefined)).to.equal(false);
+      expect(caObj.isInt(false)).to.equal(false);
+    });
+
+    it('initializeBidDefaults() should return correct object', function() {
+      let bid = caObj.initializeBidDefaults();
+      expect(Array.isArray(bid.eventCodes)).to.equal(true);
+      expect(bid.eventCodes.length).to.equal(0);
+      expect(Object.keys(bid).length).to.equal(1);
```

Comment:
Replace all 3 expect with this.

```javascript
expect(bid).to.deep.equal({'eventCodes': {}});
```

Not sure a unit test is really needed.

Reply:
Fine, I don't really trust a lot of these custom libraries to do what I expect them to do and all the documentation for them is terrible so I usually just would rather test things explicitly where I know what I'm getting.  Might as well test everything works as expected and they do a code coverage check so this will help towards that goal.

Reply:
It's a part of the [Chai Library](https://www.chaijs.com/api/bdd/) which is pretty widely used.  It makes the tests easier to read.